rvps/src/store/local_fs/mod.rs

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+//! This Store stores RV information inside a local file
+
+use std::path::Path;
+
+use anyhow::*;
+
+use crate::{ReferenceValue, Store};
+
+/// Local directory path to store the reference values,
+/// which is created by sled engine.
+const FILE_PATH: &str = "/opt/attestation-server/reference_values";
+
+/// `LocalFs` implements [`Store`] trait. And
+/// it uses rocksdb inside.
+pub struct LocalFs {
+    engine: sled::Db,
+}
+
+impl Default for LocalFs {
+    fn default() -> Self {
+        LocalFs::new(FILE_PATH).expect("Failed to create LocalFs Store.")
+    }
+}
+
+impl LocalFs {
+    /// Create a new [`LocalFs`] with given
+    /// file storage path.
+    fn new(path: &str) -> Result<Self> {
+        let path = Path::new(path);
+        let engine = sled::open(path)?;
+        Ok(Self { engine })
+    }
+}
+
+impl Store for LocalFs {
+    fn set(&mut self, name: String, rv: ReferenceValue) -> Result<Option<ReferenceValue>> {
+        let rv_serde = serde_json::to_vec(&rv)?;
+        let res = match self.engine.insert(name, rv_serde)? {
+            Some(v) => {
+                let v = serde_json::from_slice(&v)?;
+                Ok(Some(v))
+            }
+            None => Ok(None),
+        };
+
+        self.engine.flush()?;
+        res
+    }
+
+    fn get(&self, name: &str) -> Result<Option<ReferenceValue>> {
+        match self.engine.get(name)? {
+            Some(v) => {
+                let v = serde_json::from_slice(&v)?;
+                Ok(Some(v))
+            }
+            None => Ok(None),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use serial_test::serial;
+
+    use crate::{ReferenceValue, Store};
+
+    use super::{LocalFs, FILE_PATH};
+    const KEY: &str = "test1";
+
+    /// This test will test the `set` and `get` interface
+    /// for [`LocalFs`].
+    #[test]
+    #[serial]
+    fn set_and_get() {
+        {
+            let mut store = LocalFs::default();
```

Comment:
running the tests will erase `/opt/attestation-server/reference_values` ...

maybe the test should use `let mut store = LocalFs::new("test_localfs_dir")` instead of the default? (and similarly when calling remove_all_dir).
(related to all the test cases.)

Reply:
Thanks for pointing out! I'll fix this bug